### DISCUSSION TITLE
Psycopg2 dependency introduced in 1.8.0

### DISCUSSION THREAD

**Discussion by padbk** [dbt-labs/dbt-postgres#93] [General]
This [PR](https://github.com/dbt-labs/dbt-postgres/pull/60) introduced Psycopg2 as a dependecy for dbt-postgres. 

I only use dbt-postgres as it is a depedency of [dbt-redshift](https://github.com/dbt-labs/dbt-redshift/blob/80a484bb3f2faada34921adbf7a3fa8289bcf687/setup.py#L64) 

And dbt-redshift can function perfectly well without Psycopg2 as it uses [redshift-connector](https://github.com/dbt-labs/dbt-redshift/blob/80a484bb3f2faada34921adbf7a3fa8289bcf687/setup.py#L67)

Therefore when, inside a docker container, I installed dbt-redshift 1.8.0 and dbt-postgres 1.8.0, Psycopg2 was also attempted to be installed, when I have not needed it previously. This installation then failed as the docker container I was using did not have the [build prerequisites](https://www.psycopg.org/docs/install.html#build-prerequisites) to build Psycopg2 from source code.

I changed the Dockerfile to the following using an alpine image:
```
FROM python:3.11-alpine

# Psycopg2 build dependencies
RUN apk update 
RUN apk add --no-cache libpq
RUN apk add --virtual temp1 --no-cache gcc python3-dev musl-dev postgresql-dev libffi-dev

# Upgrade pip and install Poetry
RUN pip install --no-cache-dir --upgrade pip && \
    pip install --no-cache-dir poetry

# Generate the requirements file using Poetry
COPY pyproject.toml pyproject.toml
RUN poetry export --without-hashes -f requirements.txt --output requirements.txt

# Install the requirements
RUN pip install --no-cache-dir -r requirements.txt
```

This works, but I have outstanding questions:
1. Is Psycopg2 really needed as a depedency of dbt-postgres?
2. If it is needed, is there a way to skip installation of Psycopg2 when using dbt-redshift only for example?
3. If it is not possible to skip installation, is there a way to specify to use Psycopg2-binary instead of having to build from source when using Linux?

**Comment by jtcohen6** [comment]
@padbk Thanks for starting the discussion!

I left a reply on the regression report issue:
- https://github.com/dbt-labs/dbt-postgres/issues/96#issuecomment-2113020011

But I think the ultimate resolution to this question ("why do I need to install `psycopg2` in order to use `dbt-redshift`") is for us to remove `dbt-postgres` as a dependency of `dbt-redshift` — now that we've migrated to use `redshift-connector` instead, and in line with Redshift's slow-but-steady evolution away from Postgres-like syntax/functionality.
- https://github.com/dbt-labs/dbt-redshift/issues/811

**Comment by padbk** [comment]
@jtcohen6 Agree, thanks.

**Comment by ssillaots-boku** [comment]
@jtcohen6 
I've now stumbled upon this issue as well.
> But I think the ultimate resolution to this question  is for us to remove dbt-postgres as a dependency of dbt-redshift

Could you say if the resolution will be released any time soon?

**Reply by jtcohen6** [reply]
Hey @ssillaots-boku - we've started on this work, and we've identified a blocker that will require a `dbt-core` change before we can safely release this:
- https://github.com/dbt-labs/dbt-redshift/issues/811#issuecomment-2173158841

Meanwhile, we're going to revert the change to `dbt-postgres` install dependencies, which should avoid the immediate issue for most users in the short term:
- https://github.com/dbt-labs/dbt-postgres/pull/113

**Reply by ssillaots-boku** [reply]
Awesome, thanks!
